Lib/test/test_queue.py

```diff
@@ -636,6 +636,23 @@ def test_shutdown_get_task_done_join(self):
 
         self.assertEqual(results, [True]*len(thrds))
 
+    def test_shutdown_get_simple(self):
+        def get():
+            try:
+                results.append(q.get())
+            except Exception as e:
+                results.append(e)
+
+        q = self.type2test()
+        results = []
+        get_thread = threading.Thread(target=get)
+        get_thread.start()
+        q.shutdown()
```

Comment:
I suggest: `q.shutdown(immediate=False)`.

Should we add a test when shutdwon is immediate (True) ? 
And should we add a test to check pending put ?

Reply:
> The bug is specifically for getters when immediate=False, no need to increase the scope of the PR.

This seems reasonable. 
I approve the changes.